usr/lib/common/loadsave.c

```diff
@@ -2634,15 +2649,33 @@ CK_RV reload_token_object(STDLL_TokData_t *tokdata, OBJECT *obj)
 
     set_perm(fileno(fp));
 
-    if (fread(header, HEADER_LEN, 1, fp) != 1) {
+    if (fread(header, HEADER_COMMON_LEN, 1, fp) != 1) {
```

Comment:
Just realized that the same fix also needs to go into reload_token_object_old().... Will update the commit asap.

Reply:
Ah, no, reload_token_object_old() already has the logic to load public or private objects. The old data format uses the same header for both, so no further fix is needed.